fairlearn/metrics/_group_metric_set.py

```diff
@@ -93,72 +86,85 @@
 REGRESSION_METRICS[GROUP_ZERO_ONE_LOSS] = group_zero_one_loss
 
 
-def create_group_metric_set(model_type,
-                            y_true,
-                            y_preds,
-                            sensitive_features,
-                            model_titles=None,
-                            sensitive_feature_names=None,
-                            extra_metrics=None):
-    """Create a dictionary matching the Dashboard's cache."""
-    if extra_metrics is not None:
-        raise NotImplementedError("No support for extra_metrics yet")
+def _process_feature_to_integers(feature):
+    """Remap the given feature to integers indexed from 0."""
+    np_feature = _convert_to_ndarray_and_squeeze(feature)
+    unique_groups = sorted(np.unique(np_feature))
+    group_names = [str(x) for x in unique_groups]
+    groups = [unique_groups.index(x) for x in np_feature]
+    return group_names, groups
+
+
+def _process_sensitive_features(sensitive_features):
+    """Convert the dictionary into the required list."""
+    unsorted_features = []
+    for column_name, column in sensitive_features.items():
+        nxt = dict()
+        nxt[_FEATURE_BIN_NAME] = column_name
+        nxt[_BIN_LABELS], nxt[_BIN_VECTOR] = _process_feature_to_integers(column)
+        unsorted_features.append(nxt)
+    result = sorted(unsorted_features, key=lambda x: x[_FEATURE_BIN_NAME])
+    return result
+
 
-    # We could consider checking that the length of y_preds matches model_titles
-    # and that the length of sensitive_features matches sensitive_feature_names
+def _process_predictions(predictions):
+    """Convert the dictionary into two lists."""
+    names = []
+    preds = []
+    for model_name in sorted(predictions):
+        names.append(model_name)
+        y_p = _convert_to_ndarray_and_squeeze(predictions[model_name])
+        preds.append(y_p.tolist())
+    return names, preds
 
+
+def _create_group_metric_set(y_true,
```

Comment:
I think we need some sort of description of what the "dashboard's cache" expects. I suppose the doc here would be a great place for this.

Reply:
That's the doc @MiroDudik  mentions above. @rihorn2 also has a sample. But that's all we've really got for now, beyond this routine and its tests.